File: python/questions/removeDuplicates1.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        ln = len(nums)
        vp = 0
        sp = 1
        while sp < ln:
            vv = nums[vp]
            vp += 1
            nums[vp] = nums[sp]
            if nums[sp] != vv:
                sp += 1
            else:
                while sp < ln and nums[sp] == vv:
                    sp += 1
        return vp + 1

File: rust/questions/remove_duplicates1.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let nl = nums.len();
        let mut vp = 0;
        let mut sp = 1;
        while sp < nl {
            let vv = nums[vp];
            vp += 1;
            nums[vp] = nums[sp];
            if nums[sp] != vv {
                sp += 1;
            } else {
                while sp < nl && nums[sp] == vv {
                    sp += 1;
                }
            }
        }
        (vp + 1) as i32
    }
}
